docs/src/components/PlaygroundCodeEditor.tsx

```diff
@@ -1,33 +1,70 @@
-import styles from './PlaygroundCodeEditor.module.css';
-import { ChangeEvent, useState } from 'react';
+/* eslint @typescript-eslint/no-var-requires: "off" */
+
+import React, { useEffect, useRef } from 'react';
+import 'jsoneditor/dist/jsoneditor.css';
+import './jsoneditor-dark.css';
+import component_types_json_schema from '../../../schemas/component_types.schema.json';
+
+function ajvInitialization(Ajv) {
+  const ajv = Ajv({
+    allErrors: true,
+    verbose: true,
+    schemaId: 'auto',
+    $data: true,
+  });
+
+  ajv.addFormat('float', '^-?d+(.d+)?([eE][+-]?d+)?$');
+  ajv.addFormat('double', '^-?d+(.d+)?([eE][+-]?d+)?$');
+  ajv.addFormat('int32', '^-?d+$');
+  ajv.addFormat('uint32', '^d+$');
+  ajv.addFormat('uint', '^d+$');
+
+  return ajv;
+}
 
 interface PlaygroundCodeEditorProps {
   onChange: (content: object | Error) => void;
-  initialCodeEditorContent: string;
+  initialCodeEditorContent: object;
 }
 
 function PlaygroundCodeEditor({ onChange, initialCodeEditorContent }: PlaygroundCodeEditorProps) {
-  const [content, setContent] = useState<string>(initialCodeEditorContent);
-
-  const handleChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
-    const codeContent = event.target.value;
-    setContent(codeContent);
-    try {
-      const scene = JSON.parse(codeContent);
-      onChange(scene);
-    } catch (error) {
-      onChange(error);
+  const editorContainer = useRef<HTMLDivElement | null>(null);
+  const jsonEditor = useRef(null);
+
+  useEffect(() => {
+    const JSONEditor = require('jsoneditor');
```

Comment:
Instead of importing it here I would consider extracting this to separate file and conditional importing this only when running in browser